Approximate $\int_{0}^{1} xe^{-x}dx$

Composite trapezoid rule

In [11]:
import numpy as np

def f(x):
    return x*np.exp(-x)

a = 0
b = 1
n = 1
h = 1
x0 = a
x1 = b
T = (h/2)*(f(x0)+f(x1)) #trapezoid rule with two points
print("T = {}".format(T))

T = 0.18393972058572117


The composite trapezoid rule approximation with 2 points is $T=0.1839$ and the actual value of the integral is $I=\int_{0}^{1}xe^{-x}=1-2e^{-1}=0.2642$ Thus, the error is $|I-T|=0.0803$

Composite Simpson's 1/3 rule

Composite quadrature rule with two points and h = 0.05